### ISSUE TITLE
Add a way to silence stderr logging

### ISSUE THREAD

**Issue by spenczar** [matthewholman/assist#85]
I'm linking to ASSIST as a library. I have my own ways I'd like to handle errors and provide warnings to the user. It'd be nice if I could disable the `fprintf(stderr, ...)` calls in ASSIST, maybe with a `-D` flag during compilation or something.

I don't really feel that I need to be able to recover those messages in any way; just suppressing them is fine, especially if there are well-defined error codes, which is mostly the case throughout ASSIST.

**Comment by hannorein** [comment]
Fair enough. I don't think there's any harm done to adding a few `#ifdef`s. Doing anything more complicated seems like an overkill. 

(Probably not what you want, but you won't see any errors after calling `fclose(stderr)`. )
